pkg/cache/unavailableofferings_test.go

```diff
@@ -18,43 +18,193 @@ package cache
+
+	// Wait for cache expiration
+	time.Sleep(2 * time.Second)
+
+	// Test that offerings are no longer marked as unavailable after expiration
+	for _, sku := range skus {
+		assertOfferingAvailable(t, u, sku, "westus-1", karpv1.CapacityTypeOnDemand, "Offering should not be marked as unavailable after cache expiration")
+	}
+
+	// Mark the entire VM family as unavailable
+	u.MarkFamilyUnavailableWithTTL(context.TODO(), getVersionedSKUFamily(skus[0]), "westus-1", karpv1.CapacityTypeOnDemand, -1, 2*time.Second)
+
+	// Test that offerings with both more and fewer than 16 CPUs are now marked as unavailable
+	for _, sku := range skus {
+		assertOfferingUnavailable(t, u, sku, "westus-1", karpv1.CapacityTypeOnDemand, "Offering should be marked as unavailable after entire family marking")
+	}
+
+	// Test that offering from same VM family and version but in a different zone are available
+	for _, sku := range skus {
+		assertOfferingAvailable(t, u, sku, "westus-2", karpv1.CapacityTypeOnDemand, "Offering in a different zone should be available")
+	}
+
+	// Test that offerings from same VM family but different version are available
+	differentVersionSKU := createTestSKU("NV16as_v5", "NVasv5Family", 16)
+	assertOfferingAvailable(t, u, differentVersionSKU, "westus-1", karpv1.CapacityTypeOnDemand, "Offering from a different version of the same VM family should be available")
+
+	// Wait for cache expiration
+	time.Sleep(2 * time.Second)
+
+	// // Test that offerings with both more and fewer than 16 CPUs are now marked as available after expiration
+	for _, sku := range skus {
+		assertOfferingAvailable(t, u, sku, "westus-1", karpv1.CapacityTypeOnDemand, "Offering should not be marked as unavailable after cache expiration")
 	}
 }
 
 func TestUnavailableOfferings_KeyGeneration(t *testing.T) {
 	expectedKey := "spot:NV16as_v4:westus"
-	key := key("NV16as_v4", "westus", "spot")
+	key := singleInstanceKey("NV16as_v4", "westus", "spot")
 	if key != expectedKey {
 		t.Errorf("Expected key to be %s, but got %s", expectedKey, key)
 	}
 }
+
+func TestUnavailableOfferings_RestrictiveLimitPreservation(t *testing.T) {
+	// create a new cache with a long TTL to avoid expiration during test
+	singleInstanceCache := cache.New(10*time.Second, 10*time.Second)
+	vmFamilyCache := cache.New(10*time.Second, 10*time.Second)
+	u := NewUnavailableOfferingsWithCache(singleInstanceCache, vmFamilyCache)
+
+	skus := []*skewer.SKU{
+		createTestSKU("NV8as_v4", "NVasv4Family", 8),
+		createTestSKU("NV16as_v4", "NVasv4Family", 16),
+		createTestSKU("NV24as_v4", "NVasv4Family", 24),
+		createTestSKU("NV32as_v4", "NVasv4Family", 32),
+	}
+
+	// Test 1: Set a limit at 16 CPUs, then try to set a less restrictive limit at 24 CPUs
+	// The 16 CPU limit should be preserved
+	u.MarkFamilyUnavailableWithTTL(context.TODO(), getVersionedSKUFamily(skus[0]), "westus-1", karpv1.CapacityTypeOnDemand, 16, 2*time.Second)
```

Comment:
You set 10s up above but then override at 2s here... consider just hardcoding these 10s/2s times in a const for all the tests and reusing? Also possibly this should be 10s here, or there should be a variant for family that doesn't require you to specify the TTL when specifying the cores?